datafusion/core/src/datasource/listing/url.rs

```diff
@@ -56,12 +56,8 @@ impl ListingTableUrl {
     /// If the path contains any of `'?', '*', '['`, it will be considered
     /// a glob expression and resolved as described in the section below.
     ///
-    /// Otherwise, the path will be resolved to an absolute path, returning
-    /// an error if it does not exist, and converted to a [file URI]
-    ///
-    /// If you wish to specify a path that does not exist on the local
-    /// machine you must provide it as a fully-qualified [file URI]
-    /// e.g. `file:///myfile.txt`
+    /// Otherwise, the path will be resolved to an absolute path, and converted
```

Comment:
This is the change that is necessary to accomodate this, we need to permit ListingTableUrl to be created for paths that don't currently exist